### PR TITLE
sequentially assign GA to courses

### PR DESCRIPTION
**Pull request by GodwinEkuma**

### CODE REVIEW

File: GeneticAlgorithm.py [hunk 1]
```diff
@@ -282,6 +282,7 @@ def initialize(self):
                 # print("This course has a GA preference.")
                 CourseGAPref.append(course)
         # print("\n")
+        unique_labs = {}
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py:289]: Creating a dictionary for unique labs.

File: GeneticAlgorithm.py [hunk 2]
```diff
@@ -292,6 +293,12 @@ def initialize(self):
                 # print("This lab has a GA Preference.")
                 LabTAPref.append(lab)
 
+            if lab.get_labCode() in unique_labs:
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py:300]: If lab code is in unique labs, we attend it to the dictionary.

File: GeneticAlgorithm.py [hunk 3]
```diff
@@ -315,10 +322,11 @@ def initialize(self):
 
         # Iterate through all courses and labs
         # Iterate through all courses to assign a random TAGA
+        gata_course_counter = 0
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: Counter variable for courses.

File: GeneticAlgorithm.py [hunk 4]
```diff
@@ -315,10 +322,11 @@ def initialize(self):
...
         for cur_course in CourseNonGAPref:
 
-            random_gata = GAList[rnd.randrange(0, len_ga)]
-            newCourseAssignment = CourseAssignment(self._classNumb, random_gata)
+            gata = GAList[gata_course_counter]
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: Assign GAs Iteratively.

File: GeneticAlgorithm.py [hunk 5]
```diff
@@ -328,6 +336,11 @@ def initialize(self):
 
             self._assignments.append(newCourseAssignment)
 
+            if (len(GAList) - 1) == gata_course_counter:
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: If we go out of bounds for the list, initialize it back to 0, if not, add 1.

File: GeneticAlgorithm.py [hunk 6]
```diff
@@ -345,6 +358,7 @@ def initialize(self):
 
 
         # Iterate through all courses
+        gata_lab_counter = 0
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: Counter for Labs.

File: GeneticAlgorithm.py [hunk 7]
```diff
@@ -360,8 +374,8 @@ def initialize(self):
             # 
 
             # Assign a random gata
-            random_gata = TAList[rnd.randrange(0, len_ta)]
-            newCourseAssignment = CourseAssignment(self._classNumb, random_gata)
+            gata = TAList[gata_lab_counter]
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: Assign TAs iteratively.

File: GeneticAlgorithm.py [hunk 8]
```diff
@@ -371,6 +385,29 @@ def initialize(self):
 
             self._assignments.append(newCourseAssignment)
 
+            if (len(GAList) - 1) == gata_lab_counter:
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: If we go out of bounds for the TAList, initialize back to 0. If not, add 1.

File: GeneticAlgorithm.py [hunk 9]
```diff
@@ -371,6 +385,29 @@ def initialize(self):
...
+                gata_lab_counter = 0
+            else:
+                gata_lab_counter += 1
+
+        for cur_lab in unique_labs:
```
Comment (Caleb7621-java) [inline GeneticAlgorithm.py]: Go through unique labs and assign TAs to labs taught by faculty as well as TA taught lab.

File: GeneticAlgorithm.py (hunk 8 above)
Comment (Caleb7621-java) [reply GeneticAlgorithm.py]: I'm requesting changes on this comment because we should be checking if we are out of bounds of TAList and not GAList.

**Review by Caleb7621-java — DISMISSED** [review]
On line 388, change GAList to be TAList.

File: GeneticAlgorithm.py [hunk 10]
```diff
@@ -371,6 +385,29 @@ def initialize(self):
...
+
+            gata = GAList[gata_course_counter]
+            newCourseAssignment = CourseAssignment(self._classNumb, gata)
+            self._classNumb += 1
+            # Setting the course, meeting time, and semester year and append it to newClass.
+            newCourseAssignment.set_course(cur_lab)
+            newCourseAssignment.set_meetingTime(cur_lab.get_meetTimes())
```
Comment (zhao926) [inline GeneticAlgorithm.py]: This line causes an error: AttributeError: 'str' object has no attribute 'get_meetTimes'

**Review by zhao926 — CHANGES_REQUESTED** [review]

**Pull request merged**